jujugui/static/gui/src/app/components/machine-view/machine/machine.js

```diff
@@ -20,11 +20,26 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 YUI.add('machine-view-machine', function() {
 
-  juju.components.MachineViewMachine = React.createClass({
+  var dropTarget = {
+    drop: function (props, monitor, component) {
```

Comment:
This is kind of cryptic, could you add some doc blocs to these methods adding a link to the official docs.